fhirr4/ballerina/src/main/resources/fhirservice/utils.bal

```diff
@@ -52,6 +51,11 @@ isolated function getHttpService(Holder h, r4:ResourceAPIConfig apiConfig) retur
                     } else {
                         executeResourceResult = executeSearch(fhirContext, fhirService, resourceMethod);
                     }
+                    if (executeResourceResult is error) {
+                        fhirContext.setInErrorState(true);
+                        fhirContext.setErrorCode(r4:getErrorCode(executeResourceResult));
+                        return r4:handleErrorResponse(executeResourceResult);
+                    }
```

Comment:
Same error handling should happen in POST requests as well?

Reply:
yes - was focussing only on gets. thanks, fixed it.